backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java

```diff
@@ -743,8 +744,15 @@ public final void invalidateCandsFor(@NonNull final IQuery<I_C_Invoice_Candidate
 		// Schedule an update for invalidated invoice candidates
 		if (count > 0)
 		{
-			final IInvoiceCandUpdateSchedulerRequest request = InvoiceCandUpdateSchedulerRequest.of(icQuery.getCtx(), icQuery.getTrxName());
-			Services.get(IInvoiceCandUpdateSchedulerService.class).scheduleForUpdate(request);
+			icQuery.list()
+					.stream()
+					.map(I_C_Invoice_Candidate::getC_Async_Batch_ID)
+					.map(AsyncBatchId::ofRepoIdOrNone)
+					.filter(Objects::nonNull)
```

Comment:
Allright..this morning it popped into my head: 
`de.metas.invoicecandidate.api.impl.InvoiceCandDAO#invalidateCandsFor(org.compiere.model.IQuery<de.metas.invoicecandidate.model.I_C_Invoice_Candidate>)` is still a problem..
Before the change, we had one one SQL statement that inserted an arbitrary number of records into `C_Invoice_Candidate_Recompute`.
*Now*, we still do that, and then we
* execute the same query again
* load the Ics all into one list
* and iterate them

..to get the distinct C_Async_Batch_IDs and create one workpackage for each C_Async_Batch_ID.

This can cause both performance and memory problems down the road.
Plus, the data might not be the same on the second icQuery invokcation

Suggestion: 
* add a column `C_Async_Batch_ID` and a ID- (or UUID or whatever)-column to C_Invoice_Candidate_Recompute.
  * and ofc insert that ID and the ICs' C_Async_Batch_ID when doing the insert (1st icQuery invocation)
* add an `index on (that-ID, C_Async_Batch_ID)` to C_Invoice_Candidate_Recompute
* get the different C_Async_Batch_IDs with a query like `select distinct C_Async_Batch_ID from C_Invoice_Candidate_Recompute where that-ID=?`
..i.e. somthing like
```java
final List<Integer> asyncBatchIDs = Services.get(IQueryBL.class).createQueryBuilder(I_C_Invoice_Candidate_Recompute.clas)
		.addEqualsFilter(I_C_Invoice_Candidate_Recompute.COLUMNNAME_That-ID, 123)
		.create()
		.listDistinct(I_C_Invoice_Candidate_Recompute.COLUMNNAME_C_Async_Batch_ID, Integer.class);
```

With that, postgres can get the C_Async_Batch_IDs with one index scan. And it's also the same records that were added previously